src/shell/interpreter.zig

```diff
@@ -602,19 +602,233 @@ pub const EnvMap = struct {
     }
 };
 
+pub const ShellArgs = struct {
+    /// This is the arena used to allocate the input shell script's AST nodes,
+    /// tokens, and a string pool used to store all strings.
+    __arena: *bun.ArenaAllocator,
+    /// Root ast node
+    script_ast: ast.Script = .{ .stmts = &[_]ast.Stmt{} },
+
+    pub usingnamespace bun.New(@This());
+
+    pub fn arena_allocator(this: *ShellArgs) std.mem.Allocator {
+        return this.__arena.allocator();
+    }
+
+    pub fn deinit(this: *ShellArgs) void {
+        this.__arena.deinit();
+        this.destroy();
```

Comment:
This leaks the bun.ArenaAllocator pointer.


```suggestion
        bun.destroy(this.__arena);
        this.destroy();
```